### PR TITLE
Group pinned message events with MELS

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/17938

Screenshot_20210711_103649

### CODE REVIEW

File: src/utils/ReactUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import React from "react";
+
+/**
+ * Joins an array into one value with a joiner. E.g. join(["hello", "world"], " ") -> <span>hello world</span>
+ * @param array the array of element to join
+ * @param joiner the string/JSX.Element to join with
+ * @returns the joined array
+ */
+export function join(array: Array<string | JSX.Element>, joiner?: string | JSX.Element): JSX.Element {
```
Comment (turt2live) [inline src/utils/ReactUtils.tsx:25]: nit: `jsxJoin` to avoid confusing it with the array join function.

**Review by turt2live — COMMENTED** [review]
Generally seems like a good thing, though some minor code quality concerns.

**Review by turt2live — APPROVED** [review]
:shipit:

**Comment by turt2live** [comment]
except merge conflicts

**Comment by SimonBrandner** [comment]
> except merge conflicts

/me will resolve them tomorrow morning (or your evening)

**Review by turt2live — APPROVED** [review]
:shipit: :shipit: :shipit: :shipit: :shipit:

**Pull request merged**